pmd-java/src/main/java/net/sourceforge/pmd/lang/java/typeresolution/typedefinition/JavaTypeDefinitionIntersection.java

```diff
@@ -0,0 +1,183 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.java.typeresolution.typedefinition;
+
+import java.lang.reflect.Method;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+public class JavaTypeDefinitionIntersection extends JavaTypeDefinition {
+    private List<JavaTypeDefinition> intersectionTypes;
+
+    protected JavaTypeDefinitionIntersection(TypeDefinitionType defType, List<JavaTypeDefinition> intersectionTypes) {
+        super(defType);
+
+        if (intersectionTypes.isEmpty()) {
+            throw new IllegalArgumentException("Intersection type list can't be empty");
+        }
+
+        this.intersectionTypes = Collections.unmodifiableList(new ArrayList<>(intersectionTypes));
+    }
+
+    @Override
+    public Class<?> getType() {
+        return intersectionTypes.get(0).getType();
```

Comment:
there are plenty of calls to `intersectionTypes.get(0)`. I believe I understand the reason behind this, but think a proper comment is needed.

Since this is all over, maybe a private method would be in order too